### PR TITLE
Feature/yaml dto

### LINKED ISSUE #99: Get service related Kube config
As a user, I can see what Kube config MICO has created in the background for each deployed service.

Expected outcome: the resulting configuration (the yaml file) for each service that is part of an application should be presented to the user (REST endpoint: `GET /services/{shortName}/{version}/yaml` (pretty printed).
Include Kubernetes Deployment and Service.

### LINKED ISSUE #557: [UI] Show kube config in UI
As a user, I want to view the kube config yaml in the UI.
See #99.

### PR DESCRIPTION
**Pull request by Jojo134**
- [x] Ensure that the commit message is [a good commit message](https://github.com/joelparkerhenderson/git_commit_message)
- [x] Ensure to use auto format in **all** files
- [x] Tests created for changes 
- [x] Screenshots added (for UI changes)

---

## Short Description

show kubernetes yaml for service in mico-admin

image

## Resolving Issue/ Feature

Resolves #99 
Resolves #557

### CODE REVIEW

File: mico-admin/src/app/service-detail/service-detail.component.html [hunk 1]
```diff
@@ -41,9 +41,10 @@ <h3>{{service?.name}}</h3>
             <mico-service-dependency-graph [shortName]="shortName" [version]="selectedVersion">
             </mico-service-dependency-graph>
         </mat-tab>
-        <!-- DEPLOYMENT TAB-->
-        <mat-tab label="Deployment">
-            Service-Detail-Deployment...
+        <!-- KUBECONFIG TAB-->
+        <mat-tab label="kube config yaml">
```
Comment (davidkopp) [inline mico-admin/src/app/service-detail/service-detail.component.html]: Choose a different label. First letter should be in uppercase.
Proposal: Kubernetes Configuration

File: mico-core/src/main/java/io/github/ust/mico/core/dto/response/MicoYamlResponseDTO.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    /**
+     * Yaml from Kubernetes resource
+     */
+    @ApiModelProperty(extensions = {@Extension(
+        name = CustomOpenApiExtentionsPlugin.X_MICO_CUSTOM_EXTENSION,
+        properties = {
+            @ExtensionProperty(name = "title", value = "Kubernetes YAML"),
+            @ExtensionProperty(name = "readOnly", value = "true"),
+            @ExtensionProperty(name = "x-order", value = "100"),
+            @ExtensionProperty(name = "description", value = "Yaml from Kubernetes resource")
```
Comment (davidkopp) [inline mico-core/src/main/java/io/github/ust/mico/core/dto/response/MicoYamlResponseDTO.java]: YAML is an acronym. Should be in uppercase.

File: mico-core/src/main/java/io/github/ust/mico/core/service/MicoKubernetesClient.java [hunk 3]
```diff
@@ -614,6 +616,27 @@ public boolean isMicoServiceDeployed(MicoService micoService) throws KubernetesR
...
+     */
+    public String getYaml(MicoService micoService) throws KubernetesResourceException, JsonProcessingException {
+        String yaml = "";
+        Optional<Deployment> deploymentOptional = getDeploymentOfMicoService(micoService);
+        if (deploymentOptional.isPresent()) {
+            yaml += SerializationUtils.dumpWithoutRuntimeStateAsYaml(deploymentOptional.get());
+        }
+        List<Service> kubernetesServices = getInterfacesOfMicoService(micoService);
+        for (Service kubernetesService : kubernetesServices) {
+            yaml += SerializationUtils.dumpWithoutRuntimeStateAsYaml(kubernetesService);
```
Comment (davidkopp) [inline mico-core/src/main/java/io/github/ust/mico/core/service/MicoKubernetesClient.java]: Consider to use `StringBuilder.append` in loops

File: mico-core/src/main/java/io/github/ust/mico/core/resource/ServiceResource.java [hunk 4]
```diff
@@ -382,6 +384,27 @@
...
+     * @return
+     */
+    @GetMapping("/{" + PATH_VARIABLE_SHORT_NAME + "}/{" + PATH_VARIABLE_VERSION + "}" + "/yaml")
+    public ResponseEntity<Resource<MicoYamlResponseDTO>> getServiceYamlByShortNameAndVersion(@PathVariable(PATH_VARIABLE_SHORT_NAME) String shortName,
+                                                                                             @PathVariable(PATH_VARIABLE_VERSION) String version) {
+        String yaml;
+        try {
+            yaml = micoKubernetesClient.getYaml(getServiceFromDatabase(shortName, version));
+        } catch (KubernetesResourceException e) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Could not find any Service with name: '" + shortName);
```
Comment (davidkopp) [inline mico-core/src/main/java/io/github/ust/mico/core/resource/ServiceResource.java]: A `KubernetesResourceException` should not be mapped to `NOT_FOUND`. It should be `CONFLICT`. Use the message in the exception.
Proposal:
```
throw new ResponseStatusException(HttpStatus.CONFLICT, "Deployment of service '" + shortName + "' '" + version + "' has a conflict: " + e.getMessage());
```

File: mico-core/src/main/java/io/github/ust/mico/core/resource/ServiceResource.java [hunk 5]
```diff
@@ -382,6 +384,27 @@
             linkTo(methodOn(ServiceResource.class).getDependencyGraph(shortName, version)).withSelfRel()));
     }
 
+    /**
+     * Return yaml for a {@link MicoService} for the give shortname and version.
+     *
+     * @param shortName
```
Comment (davidkopp) [inline mico-core/src/main/java/io/github/ust/mico/core/resource/ServiceResource.java]: To be consistent with the other endpoints: Either set everything of the JavaDoc or nothing.

File: mico-core/src/main/java/io/github/ust/mico/core/service/MicoKubernetesClient.java [hunk 6]
```diff
@@ -614,6 +616,27 @@ public boolean isMicoServiceDeployed(MicoService micoService) throws KubernetesR
         return podList;
     }
 
+    /**
+     * Retrieves the yaml for a MicoService, contains the interfaces if they exist.
+     *
+     * @param micoService
```
Comment (davidkopp) [inline mico-core/src/main/java/io/github/ust/mico/core/service/MicoKubernetesClient.java]: Add a description of all parameters.

File: mico-admin/src/app/service-detail-kubeconfig/service-detail-kubeconfig.component.ts [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+    kubeConfigYaml: string;
+
+    constructor(
+        private apiService: ApiService,
+    ) { }
+
+    ngOnChanges() {
+        if (this.shortName != null && this.version != null) {
+
+            // TODO safeUnsubscribe subYaml
```
Comment (buehlefs) [inline mico-admin/src/app/service-detail-kubeconfig/service-detail-kubeconfig.component.ts]: unfinished todos (there are many examples already in the code...)

**Pull request merged**